circuits/src/zk_circuits/valid_relayer_fee_settlement.rs

```diff
@@ -415,3 +415,589 @@ where
+        },
+        traits::{BaseType, CircuitBaseType},
+        wallet::{Wallet, WalletShare},
+        AMOUNT_BITS,
+    };
+    use num_bigint::BigUint;
+    use rand::{thread_rng, Rng};
+    use renegade_crypto::fields::scalar_to_u128;
+
+    use crate::{
+        mpc_circuits::settle::settle_match,
+        zk_circuits::{
+            check_constraint_satisfaction,
+            test_helpers::{create_multi_opening, INITIAL_WALLET, MAX_BALANCES, MAX_ORDERS},
+            valid_relayer_fee_settlement::test_helpers::create_witness_statement,
+            valid_wallet_create::SizedValidWalletCreateStatement,
+        },
+    };
+
+    use super::{
+        ValidRelayerFeeSettlement, ValidRelayerFeeSettlementStatement,
+        ValidRelayerFeeSettlementWitness,
+    };
+
+    // -----------
+    // | Helpers |
+    // -----------
+
+    /// The Merkle height used for testing
+    const MERKLE_HEIGHT: usize = 5;
+    /// A statement type with the testing constants attached
+    type SizedStatement = ValidRelayerFeeSettlementStatement<MAX_BALANCES, MAX_ORDERS>;
+    /// A witness type with the testing constants attached
+    type SizedWitness = ValidRelayerFeeSettlementWitness<MAX_BALANCES, MAX_ORDERS, MERKLE_HEIGHT>;
+
+    /// Get a pair of initial wallets to test the circuit with
+    fn get_initial_wallets() -> (Wallet<MAX_BALANCES, MAX_ORDERS>, Wallet<MAX_BALANCES, MAX_ORDERS>)
+    {
+        let mut rng = thread_rng();
+        let mut sender_wallet = INITIAL_WALLET.clone();
+        let mut recipient_wallet = INITIAL_WALLET.clone();
+        sender_wallet.blinder = Scalar::random(&mut rng);
+        recipient_wallet.blinder = Scalar::random(&mut rng);
+
+        (sender_wallet, recipient_wallet)
+    }
+
+    /// Return whether the constraints are satisfied for the given witness and
+    /// statement
+    fn check_constraints_satisfied(statement: &SizedStatement, witness: &SizedWitness) -> bool {
+        check_constraint_satisfaction::<
+            ValidRelayerFeeSettlement<MAX_BALANCES, MAX_ORDERS, MERKLE_HEIGHT>,
+        >(witness, statement)
+    }
+
+    // ---------
+    // | Tests |
+    // ---------
+
+    /// Tests constraint satisfaction on a valid witness and statement
+    #[test]
+    fn test_valid_witness() {
+        let (sender_wallet, recipient_wallet) = get_initial_wallets();
+        let (statement, witness) = create_witness_statement(&sender_wallet, &recipient_wallet);
+
+        assert!(check_constraints_satisfied(&statement, &witness));
+    }
+
+    /// Test the case in which a fee is settled into a wallet with zero'd
+    /// balances
+    #[test]
+    fn test_valid_witness__zero_initial_receiver_balance() {
+        let (sender_wallet, mut recipient_wallet) = get_initial_wallets();
+        for bal in recipient_wallet.balances.iter_mut() {
+            *bal = Balance::default();
+        }
+
+        let (statement, witness) = create_witness_statement(&sender_wallet, &recipient_wallet);
```

Comment:
the zero mint in the default balance on the recipient wallet gets overwritten in `create_witness_statement`, so this won't test the settlement of a fee into a balance with a previously zero mint